src/components/appheader.main.jsx

```diff
@@ -187,7 +193,7 @@ class AppHeaderMain extends React.Component {
         </div>
 
         <div className="collapsable-container collapse collapsed">
-          {(!Config.b2b.enable || (Config.b2b.enable && isB2bCartSelected)) && (
+          {(!Config.b2b.enable || (Config.b2b.enable && isB2bCartSelected && availability)) && (
```

Comment:
We shouldn't hide the search altogether here. The CATALOG_BROWSER role has access to browse the catalog, but cannot add to cart or wishlists. So on the Product Display Page, we should disable those buttons if the forms are not returned by the API.